### DISCUSSION TITLE
Use diligent not in cmake

### DISCUSSION THREAD

**Discussion by yillis** [DiligentGraphics/DiligentEngine#276] [General]
I am experiencing several issues with Diligent Engine when not using it under CMake. I compiled Diligent Engine into a DLL through CMake's install target and encountered the following problems:

- After compiling the Tool, there are no header files for NativeApp. It seems that NativeApp is not included in the install compilation.

- For instance, the header file directory is include/DiligentTool/ImGui/xxx. In these header files, third-party libraries like ImGui are included directly using #include "Imgui.h". This causes issues in my build system as it cannot find these header files, making the resolution process somewhat cumbersome. Similarly, many other files directly include headers from different directories. I need to add directories like DiligentCore/GraphicsEngine/interface/ to my include dir individually. Is there a more efficient solution to this?

**Comment by TheMostDiligent** [comment]
The install currently works for the Core module. Other modules are currently intended to be used with CMake at the moment.
It can certainly be improved though.

**Reply by yillis** [reply]
Understood. I will focus on using core first, and then manually build the required tools into my project. In addition, do you recommend using the native-app in the Tool, or the combination of glfw+diligent? Thank you.

**Reply by TheMostDiligent** [reply]
> the native-app in the Tool, or the combination of glfw+diligent

It is totally up to you really. There is an [example](https://github.com/DiligentGraphics/DiligentSamples/tree/master/Samples/GLFWDemo) that shows how to use GLFW with Diligent.
